ir_datasets/datamaestro/data.py

```diff
@@ -0,0 +1,52 @@
+from pathlib import Path
+from typing import Iterator
+from datamaestro_text.interfaces.trec import parse_qrels
+from experimaestro import Config, Option
+import datamaestro_text.data.ir as ir
+import ir_datasets
+
+# Interface between ir_datasets and datamaestro:
+# provides adapted data types
+
+
+class IRDSId(Config):
+    @classmethod
+    def __xpmid__(cls):
+        return f"ir_datasets.{cls.__qualname__}"
+
+    irds: Option[str]
+
+
+class AdhocTopics(ir.AdhocTopics, IRDSId):
+    def iter(self) -> Iterator[ir.AdhocTopic]:
+        """Returns an iterator over topics"""
+        ds = ir_datasets.load(self.irds)
+        for query in ds.queries_iter():
+            yield ir.AdhocTopic(query.query_id, query.text)
```

Comment:
There's [an issue](https://github.com/allenai/ir_datasets/issues/72) to make sure every query and document has a field called "text", but right now, that's not necessarily the case. The original design was to have the names in IRDS mimic the field names used by the original datasets when available. This involves a judgment call on what the appropriate action is, but I think it usually should be clear.

If it would be helpful for this integration, I could prioritize that issue.

Reply:
Yes this would be great since this would allow having a generic method for the basic case, thanks